client/finality-grandpa/src/observer.rs

```diff
@@ -370,3 +370,84 @@ where
 		Future::poll(Pin::new(&mut self.network), cx)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	use crate::{aux_schema,	communication::tests::{Event, make_test_network}};
+	use substrate_test_runtime_client::{TestClientBuilder, TestClientBuilderExt};
+	use sc_network::PeerId;
+
+	use futures::executor::{self, ThreadPool};
+
+	#[test]
+	/// Ensure `Future` implementation of `ObserverWork` is polling its `NetworkBridge`. Regression
```

Comment:
sure we need doc comments for test function?

Reply:
Why not?

Sometimes an explanations helps on a refactor.

Reply:
I mean they can be just comments, not doc comments

Reply:
But I don't really mind